### PR TITLE
Stop attempting template substitutions for invalid files

### LINKED ISSUE #204: Do not attempt template substitution on files that cannot be parsed by liquid
liquid fails to parse some files and if that's the case, cargo-generate shouldn't try to look for template strings.

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**
Fixes #204

### CODE REVIEW

**Comment by EverlastingBugstopper** [comment]
@k0pernicus do you think we could do a point release for this?

**Comment by k0pernicus** [comment]
@EverlastingBugstopper I was waiting for another review, but I can merge this fix tomorrow if no another review has been added here :ok_hand:

**Comment by EverlastingBugstopper** [comment]
I'd also love another review on this, always good to have that.

File: tests/integration/basics.rs [hunk 1]
```diff
@@ -40,6 +40,38 @@ version = "0.1.0"
         .contains("foobar-project"));
 }
 
+#[test]
+fn it_skips_substitution_for_random_garbage_in_cargo_toml() {
+    let template = dir("template")
+        .file(
+            "Cargo.toml",
+            r#"[package]
+name = "{{function fart() { return "pfffttt"; } fart();}}"
```
Comment (EverlastingBugstopper) [inline tests/integration/basics.rs]: huehuehuehue

Comment (EverlastingBugstopper) [reply tests/integration/basics.rs]: pfffttt

**Comment by ashleygwilliams** [comment]
Screen Shot 2019-11-21 at 3 40 09 PM

**Comment by ashleygwilliams** [comment]
OK so

with the last commit, the output looks like this 
Screen Shot 2019-11-21 at 3 51 52 PM

i dont love this tho. this could mask actually errors... especially with a future facing perspective on the idea that we'll implement custom variables. thoughts @k0pernicus @EverlastingBugstopper ?

**Comment by EverlastingBugstopper** [comment]
perhaps a better approach is to look at the contents of the template placeholder and make sure it matches a name we expect like "template_name" or "author"

**Comment by ashleygwilliams** [comment]
@EverlastingBugstopper the whitelist approach makes sense. ill see what i can do!

**Comment by jhpratt** [comment]
@ashleygwilliams Any chance we can get something along the lines of this merged? It's basically impossible to have GitHub Actions integrated in a template currently.

**Comment by k0pernicus** [comment]
@EverlastingBugstopper @ashleygwilliams any update on this please? :)

**Comment by EverlastingBugstopper** [comment]
unfortunately i dont have bandwidth for this right now - sorry :(

**Comment by Veetaha** [comment]
Basically this depends on @ashleygwilliams, if you feel you don't have time/motivation for this project, I recommend passing the rights to merge and publish releases to crates.io to some most active contributor

**Comment by sassman** [comment]
> impossible to have GitHub Actions integrated in a template currently.
This is already possible (on master) with the use of (raw tags)[https://shopify.github.io/liquid/tags/raw/].

But other than that, if there are like for instance unknown variables used, it still fails. So a solution to this issue is still appreciated I guess. But I lost a bit track of what needs to be done to get it merged, @k0pernicus any idea what needs to be done?

**Comment by k0pernicus** [comment]
> This is already possible (on master) with the use of raw tags.

Totally.

> But other than that, if there are like for instance unknown variables used, it still fails. So a solution to this issue is still appreciated I guess. But I lost a bit track of what needs to be done to get it merged, @k0pernicus any idea what needs to be done?

To be honest I lost the track too...
I can check the changes and reviews to "refresh my mind" in the next days, and rebase the work from @EverlastingBugstopper to match our master.

**Comment by sassman** [comment]
> I can check the changes and reviews to "refresh my mind" in the next days, and rebase the work from @EverlastingBugstopper to match our master.

No worries, I was working recently on a fail-safe handling as well, so I can double-check this one and rebase it too. I have a bit of time this weekend.

**Comment by k0pernicus** [comment]
Awesome, thanks @sassman :+1: :)

**Comment by sassman** [comment]
just for cross-reference there is now PR #245 that aims to tackle the thing and also handling undefined variables issues. So the scope is a bit wider than this one. Feel free to discuss pros and cons there.

**Pull request closed without merging**